### PR TITLE
Group load

### PR DESCRIPTION
**Pull request by vigji**
Here it is! 
In this PR, I addressed:
 - #96 by making the column from which infos are loaded either "group" or "KSLabel" [here](https://github.com/iurillilab/pynapple/blob/c46d80b00abbc38a3782aa4a6069b8454a157817/pynapple/io/phy.py#L177)
 - In doing so, I refactored a bit the loading function, it had some brittle parts. I made `cluster_info`/`cluster_group` only `cluster_info`, and not making it an attribute as it is set and used only in the method to load from Phy folder.
 - There's also a minor fix in the rates calculation, for some reason with my very ugly data I have neurons with no spikes and was getting time supports of 0 duration; now frequency is set to np.nan in those cases. I think in general is a good fallback but happy to remove it if it is too specific for my edge case
 - I implemented what I suggested in #97: now spikes are all loaded, regardless of their classification, in the self._all_spikes attribute; and then selected (with caching, as the `TsGroup` indexing seems slow, 1-2s) when querying the self.spikes property, depending on what is specified in the `valid_group_labels` argument (by default `["good"]`). Labels are stored as a "label" unit attribute in the nab file. Back compatibility is ensued by the fact that if no label is found in the file, the filtering is just always returning all spikes.

Locally, tests are running. I don't find specifications for linting (are you considering implementing it? I'd recommend it!) so I don't know what to adhere to formatting-wise, feel free to let me know in case.
I will make sure I clean up prints and notebooks changes.
Let me know! And thank you for the package!

### CODE REVIEW

**Comment by vigji** [comment]
Also, tests are running locally on python 3.9. I still have to implement a specific test for this case, let me know if you're happy with me adding another file to the tests assets to deal with both possible scenarios of data with and without labels. 
Would you consider implementing GitHub Actions to automatise the rest of the testing in other python environments?

**Comment by gviejo** [comment]
Hello Luigi, Thanks for the PR. I will test it as soon as possible next week.

**Comment by gviejo** [comment]
Hey Luigi,
I wrote one comment about something I am not sure works. Maybe it's a problem of phy version.

I like the idea of caching the label good but I am not sure it provides enough flexibility. Looking at this issue, I am starting to think that the best is just to add all the metainfo available in the TsGroup and let the user decide after what to pick with groupby_category for example :
```
spikes = data.spikes.getby_category("KSLabel")['good']
```
or
```
mua = data.spikes.getby_category("UserLabel")["mua"]
```
So basically, the object data.spikes should have the following columns 
[(freq', KSLabel, UserLabel, shank, channel) (skipping them if it's not found in the cluster_info)


Ok for the rate fix otherwise. And for Github Actions, I need to look into it.

**Comment by gviejo** [comment]
Hey Luigi, 

Here is the review I wrote :

image

But I think it doesn't matter now that we agree on returning both the KSLabel and the UserLabel for all units. 
Let me know if you want to make a new PR or if you want to update this one? I can also make the modifications myself.

And yes for the PR for github actions. If you can write somewhere a short description of what each file does, that would be great.
Thanks
Guillaume

**Comment by vigji** [comment]
> But I think it doesn't matter now that we agree on returning both the KSLabel and the UserLabel for all units. Let me know if you want to make a new PR or if you want to update this one? I can also make the modifications myself.

All makes sense to me! I would keep it into this PR if it is fine for you, as there's already all changes related to loading those infos in the units file. So, should I just transfer all columns that can be find in the info file to new attributes of the units?

> And yes for the PR for github actions. If you can write somewhere a short description of what each file does, that would be great.

Great, than I would do another PR for that, I'll open an issue for the details!

**Comment by gviejo** [comment]
Hello Luigi, what is the status of this PR? I think we agreed to load both the KSLabel and the UserLabel.

**Comment by vigji** [comment]
I am happy to push this forward, but I need your feedback on the part that would change behavior of the code for current users. In the scenario we are envisioning `data.spikes` should return all available units I guess, but this will change the behavior of existing scripts that just use `data.spikes` with folders for which there is no `.nwb` file yet. Is this ok?

**Comment by gviejo** [comment]
I need to check if there is a compartment in nwb to store and retrieve the KSLabel and UserLabel. Otherwise, I agree it would be hard to have the same information in data.spikes if you load from nwb versus you load from the phy files.

**Comment by vigji** [comment]
No wait, that is not exactly what I meant: what I was raising is that, assuming that there is a way to store KSLabel and UserLabel in the nwb (it should be flexible enough to do it), then we have to deal with the default behavior of `data.spikes`. I would assume that it should return all spike stored, while `data.spikes.getby_category("UserLabel")["x"]` should filter spikes given some labels. But this means that all the existing code where `data.spikes` is used will change its behavior, this is what I am concerned about. The behavior won't change when working on a nwb file which was already generated filtering all "good" spikes, I agree, but still, maybe we want to raise a warn?

**Comment by gviejo** [comment]
I think I see your point. And I think the warning solution is a good idea for already existing code.

**Comment by gviejo** [comment]
Hey Luigi, I move the IO to [nwbmatic](https://github.com/pynapple-org/nwbmatic) as discussed in #130. Is it possible to close this PR and open a new one in nwbmatic if this is still an issue?

**Comment by vigji** [comment]
Sorry, I thought I got back on this already! Sure, will test it out soon.
